tailscale/provider.go

```diff
@@ -84,14 +91,18 @@ func Provider(options ...ProviderOption) *schema.Provider {
 		},
 	}
 
+	provider.ConfigureContextFunc = func(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
```

Comment:
Moved setting `ConfigureContextFunc` here so we can pass the provider into `providerConfigure`.